core/commands/add.go

```diff
@@ -180,26 +182,35 @@ You can now check what blocks have been created by:
 		//	return
 		//}
 
-		progress, _, _ := req.Option(progressOptionName).Bool()
-		trickle, _, _ := req.Option(trickleOptionName).Bool()
-		wrap, _, _ := req.Option(wrapOptionName).Bool()
-		hash, _, _ := req.Option(onlyHashOptionName).Bool()
-		hidden, _, _ := req.Option(hiddenOptionName).Bool()
-		silent, _, _ := req.Option(silentOptionName).Bool()
-		chunker, _, _ := req.Option(chunkerOptionName).String()
-		dopin, _, _ := req.Option(pinOptionName).Bool()
-		rawblks, rbset, _ := req.Option(rawLeavesOptionName).Bool()
-		nocopy, _, _ := req.Option(noCopyOptionName).Bool()
-		fscache, _, _ := req.Option(fstoreCacheOptionName).Bool()
-		cidVer, _, _ := req.Option(cidVersionOptionName).Int()
-		hashFunStr, hfset, _ := req.Option(hashOptionName).String()
+		progress, _ := req.Options[progressOptionName].(bool)
+		trickle, _ := req.Options[trickleOptionName].(bool)
+		wrap, _ := req.Options[wrapOptionName].(bool)
+		hash, _ := req.Options[onlyHashOptionName].(bool)
+		hidden, _ := req.Options[hiddenOptionName].(bool)
+		silent, _ := req.Options[silentOptionName].(bool)
+		chunker, _ := req.Options[chunkerOptionName].(string)
+		dopin, _ := req.Options[pinOptionName].(bool)
+		rawblks, rbset := req.Options[rawLeavesOptionName].(bool)
+		nocopy, _ := req.Options[noCopyOptionName].(bool)
+		fscache, _ := req.Options[fstoreCacheOptionName].(bool)
+		cidVer, _ := req.Options[cidVersionOptionName].(int)
+		hashFunStr, hfset := req.Options[hashOptionName].(string)
 
 		if nocopy && !cfg.Experimental.FilestoreEnabled {
 			res.SetError(errors.New("filestore is not enabled, see https://git.io/vy4XN"),
 				cmdkit.ErrClient)
 			return
 		}
 
+		if hfset && hashFunStr != "sha2-256" && cidVer == 0 {
+			cidVer = 1
```

Comment:
This should be:

```go
if hfset && hashFunStr != "sha2-256" && cidVer == 0 {
    if cidVerSet {
        res.SetError(...)
        return
    }
    cidVer = 1
}
```

Reply:
Hrm... thats a difficult thing now. 

We don't have an easy way to distinguish between default values and unset values.

Reply:
We can just not set a default value. However, we should probably just do this in a follow up PR.

Reply:
I just went ahead and fixed this anyways.